### ISSUE TITLE
Rule proposal: no spread single property

### ISSUE THREAD

**Issue by rightaway** [sindresorhus/eslint-plugin-unicorn#2017]
It can check that spread isn't used when there's only a single property. So `{ ...expression }` and `[ ...expression ]` should both just be `expression`.

Sometimes spread is used just to combine things, so if you start off with `{ ...a, b: null, c: null }` and then later remove `b` and `c` you're left with `{ ...a }` when you really could replace it with `a`.

@fisker https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1482#issuecomment-1320820629.

**Comment by silverwind** [comment]
I often use `{...expression}` as shallow clone.

**Comment by fisker** [comment]
> thought I agree Object.assign({}, obj) may be more clear.

I prefer `{...obj}`.

#1482 mean to report cases that unnecessary cloning an array, to align with that it should only report unnecessary cloning an object. But the only thing in my mind is `{...Object.create(...)}` and `{...Object.fromEntries(...)}`.

**Comment by silverwind** [comment]
I sometimes do defensive clones with that syntax, but I guess those may all as well be deep clones (using [structuredClone](https://developer.mozilla.org/en-US/docs/Web/API/structuredClone) ideally), so maybe the rule could still be useful to me.

**Comment by rightaway** [comment]
> I think this should only check the spreading object is known an object, otherwise there will be too many false positives.

@fisker Can you elaborate some examples where it would be false positive?

**Comment by fisker** [comment]
> @fisker Can you elaborate some examples where it would be false positive?

```js
const a = undefined;
const b = {...a};

// vs

const a = undefined;
const b = a;
```

**Comment by fregante** [comment]
This sounds similar to:

- https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-useless-spread.md

but it conflicts with at least two rules:

- https://eslint.org/docs/latest/rules/prefer-object-spread
- https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-spread.md

While it kind of makes sense, `{...z}` and `[...a]` are common ways to have shallow copies. I would not use this rule

**Comment by sindresorhus** [comment]
Closing as not planned. Single spread of an unknown value is often intentional shallow cloning or object normalization, so reporting `{...value}` or `[...value]` would produce too many false positives and can change behavior. `no-useless-spread` should stay limited to literal spreads and known redundant clones.
